src/cmd/initialize.go

```diff
@@ -73,6 +81,62 @@ var initCmd = &cobra.Command{
 	},
 }
 
+// DEPRECATED_V1.0.0: --nodeport should be removed from the cli in v1.0.0
+func setRegistryNodePort() {
+	configVar := "REGISTRY_NODEPORT"
+
+	internalRegistry := pkgConfig.InitOpts.RegistryInfo.Address == ""
+
+	// warn for deprecation
+	if nodePortArg != 0 {
+		message.Warn(lang.WarnNodePortDeprecated)
+	}
+
+	// check the --set REGISTRY_NODEPORT first
+	configuredNodePort, err := strconv.Atoi(pkgConfig.PkgOpts.SetVariables[configVar])
+	if err != nil {
+		configuredNodePort = 0
+	}
+
+	// check the old --nodeport flag second
+	if configuredNodePort == 0 {
+		configuredNodePort = nodePortArg
+	}
+
+	// the user can't set both that this is an external registry and the nodeport stuff.
+	if !internalRegistry && configuredNodePort != 0 {
+		message.Fatal(nil, "both --registry-url and --nodeport are set, please only use one")
+	}
+
+	if internalRegistry {
+		if configuredNodePort > 32767 || configuredNodePort < 30000 {
+			configuredNodePort = config.ZarfInClusterContainerRegistryNodePort
+		}
+		pkgConfig.PkgOpts.SetVariables[configVar] = strconv.Itoa(configuredNodePort)
+		pkgConfig.InitOpts.RegistryInfo.Address = fmt.Sprintf("%s:%d", helpers.IPV4Localhost, configuredNodePort)
+		pkgConfig.InitOpts.RegistryInfo.NodePort = configuredNodePort
+	} else {
+		// do not set the nodeport if this is an external registry
+		pkgConfig.PkgOpts.SetVariables[configVar] = ""
```

Comment:
We should be able to remove this else block. If the user specifies an external registry, we shouldn't have to set the `REGISTRY_NODEPORT` var to an empty string